kingpin/actors/aws/cloudformation.py

```diff
@@ -0,0 +1,434 @@
+                aws_settings.AWS_SECRET_ACCESS_KEY):
+            raise exceptions.InvalidCredentials(
+                'AWS settings imported but not all credentials are supplied. '
+                'AWS_ACCESS_KEY_ID: %s, AWS_SECRET_ACCESS_KEY: %s' % (
+                    aws_settings.AWS_ACCESS_KEY_ID,
+                    aws_settings.AWS_SECRET_ACCESS_KEY))
+
+        self.conn = cloudformation.connect_to_region(
+            self.option('region'),
+            aws_access_key_id=aws_settings.AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=aws_settings.AWS_SECRET_ACCESS_KEY)
+
+    @concurrent.run_on_executor
+    @retry(retry_on_exception=retry_if_transient_error,
+           stop_max_attempt_number=5,
+           wait_exponential_multiplier=500,
+           wait_exponential_max=aws_settings.CF_WAIT_MAX)
+    @utils.exception_logger
+    def _get_stacks(self):
+        """Gets a list of existing CloudFormation stacks.
+
+        Gets a list of all of the stacks currently in the account, that are not
+        in the status 'DELETE_COMPLETE'.
+
+        Returns:
+            A list of boto.cloudformation.stack.StackSummary objects.
+        """
+        # Get the list of all possible stack statuses from the Boto module,
+        # then pull out the few that indicate a stack is no longer in
+        # existence.
+        self.log.debug('Getting list of stacks from Amazon..')
+        statuses = list(self.conn.valid_states)
+        statuses.remove('DELETE_COMPLETE')
+        return self.conn.list_stacks(stack_status_filters=statuses)
+
+    @gen.coroutine
+    def _get_stack(self, stack):
+        """Returns a cloudformation.Stack object of the requested stack.
+
+        Args:
+            stack: String name
+
+        Returns
+            <Stack Object> or <None>
+        """
+        stacks = yield self._get_stacks()
+        self.log.debug('Checking whether stack %s exists.' % stack)
+        new_list = [s for s in stacks if s.stack_name == stack]
+
+        if len(new_list) > 0:
+            raise gen.Return(new_list[0])
+
+        raise gen.Return(None)
+
+    @gen.coroutine
+    def _wait_until_state(self, desired_states, sleep=15):
+        """Indefinite loop until a stack has finished creating/deleting.
+
+        Whether the stack has failed, suceeded or been rolled back... this
+        method loops until the process has finished. If the final status is a
+        failure (rollback/failed) then an exception is raised.
+
+        Args:
+            desired_states: (tuple/list) States that indicate a successful
+                            operation.
+            sleep: (int) Time in seconds between stack status checks
+
+        Raises:
+            StackNotFound: If the stack doesn't exist.
+        """
+        while True:
+            stack = yield self._get_stack(self.option('name'))
+
+            if not stack:
+                msg = 'Stack %s not found?' % self.option('name')
```

Comment:
This shouldn't be a question. Would also suggest wrapping %s as `"%s"` so that the message is never "Stack not found"